closingd/closingd.c

```diff
@@ -609,13 +649,21 @@ int main(int argc, char *argv[])
 	/* stdin == requests, 3 == peer, 4 = gossip, 5 = gossip_store, 6 = hsmd */
 	per_peer_state_set_fds(pps, 3, 4, 5);
 
+	snprintf(fee_negotiation_step_str, sizeof(fee_negotiation_step_str),
+		 "%" PRIu64 "%s", fee_negotiation_step,
+		 fee_negotiation_step_unit ==
+			 CLOSING_FEE_NEGOTIATION_STEP_UNIT_PERCENTAGE
+		     ? "%"
+		     : "sat");
+
```

Comment:
tal_fmt() is generally safer than snprintf, BTW.

Reply:
It was using `tal_fmt()` in a [previous incarnation](https://github.com/ElementsProject/lightning/commit/662c351a826fe01d91c92df1e4bbc86665871d50#diff-9d466ee4dfc90b4d2dbf7958605b7d4aR642) of this PR. However, that ended up with an obscure use-after-free:

* create the string with `tal_fmt()` from `tmpctx`
* print the string with `status_debug()`
* call `do_reconnect()` which calls `closing_read_peer_msg()` which calls `clean_tmpctx()` which frees `tmpctx`
* try to print the string with `peer_billboard()`, but it is already freed